### PR TITLE
Add Data preprocessing with ONNX: ResNet-50 example

### PR DESCRIPTION
**Pull request by jantonguirao**
Adds a tutorial on how to use ONNX to represent a data preprocessing graph and combine it with an existing network. The tutorial uses ResNet-50 as an example.

Signed-off-by: Joaquin Anton <janton@nvidia.com>

### CODE REVIEW

File: tutorials/PreprocessingResnet50.ipynb [hunk 1]
```diff
@@ -0,0 +1,675 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "84ae6f04",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "url = 'https://raw.githubusercontent.com/onnx/models/main/vision/classification/resnet/model/resnet50-v1-12.onnx'\n",
+    "out_file = 'resnet50-v1-12.onnx'\n",
+    "download_lfs_file(url, out_file)"
```
Comment (jcwchen) [inline tutorials/PreprocessingResnet50.ipynb:154]: nit: we can directly use onnx.hub to get the model
```
from onnx import hub
rn50_network = hub.load("ResNet50-fp32")
```

Comment (jantonguirao) [reply tutorials/PreprocessingResnet50.ipynb]: That's great. Done

File: tutorials/PreprocessingResnet50.ipynb [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "96417141",
+   "metadata": {},
+   "source": [
+    "## Using an existing model\n",
+    "\n",
+    "For this example, we will use an existing ResNet50 model from the ONNX model repository. Let us load and inspect the model to see what inputs/outputs it expects"
```
Comment (jcwchen) [inline tutorials/PreprocessingResnet50.ipynb:92]:
```suggestion
    "For this example, we will use an existing ResNet50 model from the ONNX model repository. Let us load and inspect the model to see what inputs/outputs it expects."
```

File: tutorials/PreprocessingResnet50.ipynb [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2023-01-25 13:49:38.308705906 [W:onnxruntime:, graph.cc:3490 CleanUnusedInitializersAndNodeArgs] Removing initializer '_0x481a1c0__0x4a02f20_axes_input'. It is not used by any node and should be removed from the model.\n",
+      "2023-01-25 13:49:38.308739102 [W:onnxruntime:, graph.cc:3490 CleanUnusedInitializersAndNodeArgs] Removing initializer '_0x481a1c0__0x4a02f20_k2'. It is not used by any node and should be removed from the model.\n"
+     ]
+    }
+   ],
+   "source": [
+    "session = onnxruntime.InferenceSession('rn50_preprocessing.onnx', None)\n",
```
Comment (jcwchen) [inline tutorials/PreprocessingResnet50.ipynb:245]:
```suggestion
    "session = onnxruntime.InferenceSession('rn50_preprocessing.onnx', providers=["CPUExecutionProvider"])\n",
```

**Review by jcwchen — APPROVED** [review]
Some nit comments. Thanks!

File: tutorials/PreprocessingResnet50.ipynb (hunk 3 above)
Comment (jantonguirao) [reply tutorials/PreprocessingResnet50.ipynb]: Done

File: tutorials/PreprocessingResnet50.ipynb (hunk 2 above)
Comment (jantonguirao) [reply tutorials/PreprocessingResnet50.ipynb]: Done

**Comment by jcwchen** [comment]
Thank you for the quick updates. I will merge it after onnxruntime 1.14 has been released (which finally supports opset_version 18).

**Review by gramalingam — APPROVED** [review]
Looks great, thanks for the tutorial! (Minor nit: I think even the model-construction can use the parser to simplify it, just like in the case of the graph. Just to encourage others also to use it.)

**Comment by jantonguirao** [comment]
@gramalingam I adjusted the tutorial to use the parser API. It looks much better now, thanks.

**Comment by jcwchen** [comment]
ORT 1.14 has been released. I will merge this PR. Thanks @jantonguirao for adding!

**Pull request merged**